api/envoy/data/core/v3/health_check_event.proto

```diff
@@ -66,6 +67,12 @@ message HealthCheckEvent {
 
   // Timestamp for event.
   google.protobuf.Timestamp timestamp = 6;
+
+  // Host metadata
+  config.core.v3.Metadata metadata = 10;
+
+  // Host locality
+  config.core.v3.Locality locality = 11;
```

Comment:
If there is no transport socket matcher in the config, it will use the one same as the cluster one. It makes sense to specify the http codec type h1/h2/h3.